src/color.h

```diff
@@ -400,6 +400,25 @@ class color_manager : public JsonSerializer, public JsonDeserializer
 
 color_manager &get_all_colors();
 
+/**
+ * For color values that are created *before* the color definitions are loaded
+ * from JSON. One can't use the macros (e.g. c_white) directly as they query
+ * the color_manager, which may not be initialized. Instead one has to use
+ * the color_id (e.g. def_c_white) and translate the id to an actual color
+ * later. This is done by this class: it stores the id and translates it
+ * when needed to the color value.
+ */
+class deferred_color
```

Comment:
It's not working, I just tried it. The result is the same, only with different color ids - but still a debug message and all fields have the same white color in the editmap menu.

The base problem is that the macro `c_white` resolves to `get_all_colors().get(def_c_white)`, which calls `color_manager::get`. But the manager loads the color definitions *after* `main` has been entered, which is *after* the initialization of static objects. So it basically returns values from an uninitialized `std::array<> color_array`, which get stored in `fieldlist`.

Anyway, I tested this solution and it works fine. It even allows to change the color at runtime via the color manager (e.g. you can change `c_red` to mean blue in the color manager and all blood tiles will appear blue).